parachain/pallets/verifier/src/lib.rs

```diff
@@ -93,12 +156,71 @@ impl<T: Trait> Module<T> {
 	}
 
 	// Verify that the message sender matches the relayer account
-	fn verify_sender(sender: T::AccountId) -> DispatchResult {
-		if sender != RelayKey::<T>::get() {
+	fn verify_sender(sender: &T::AccountId) -> DispatchResult {
+		if *sender != RelayKey::<T>::get() {
 			return Err(Error::<T>::Invalid.into())
 		}
 		Ok(())
 	}
+
+	// Validate an Ethereum header for import
+	fn validate_header_to_import(sender: &T::AccountId, header: &EthereumHeader) -> Result<(), &'static str> {
+		Self::verify_sender(sender)?;
+
+		ensure!(
+			EthHeaders::<T>::contains_key(header.parent_hash),
+			"Parent header must be imported first",
```

Comment:
This error handling could be improved a bit. Instead of returning a free-form string as a error, we could define a custom error type for the module. This aids in documentation and runtime introspection. For a primer see https://substrate.dev/docs/en/knowledgebase/runtime/errors.

In your case, we could add `MissingParentHeader` and `DuplicateHeader` errors:
```
decl_error! {
    pub enum Error for Module<T: Trait> {
        /// Verification scheme is not supported.
        NotSupported,
        /// The message failed verification.
        Invalid
        /// Missing parent header
        MissingParentHeader
        /// DuplicateHeader
        DuplicateHeader
    }
}
```
And then use them like this:
```
ensure!( ... , Error::<T>::MissingParentHeader );
```